pkg/cloud/libvirt/client/client.go

```diff
@@ -559,7 +565,14 @@ func (client *libvirtClient) GetDHCPLeasesByNetwork(networkName string) ([]libvi
 		return nil, err
 	}
 
-	return network.GetDHCPLeases()
+	dhcpLeases, err := network.GetDHCPLeases()
+	if err != nil {
+		glog.Errorf("Network %s failed to return dhcp leases", networkName)
+		return nil, err
+	}
+	network.Free()
```

Comment:
should we just do a `defer network.Free()` when we get the network object above after the nil sanity check ?

Reply:
oh yeah, good call.  the original patch doesnt cover the second error case